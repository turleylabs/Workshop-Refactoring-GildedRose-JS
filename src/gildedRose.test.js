import { GildedRose, Item } from "./gildedRose";
import product from 'cartesian-product';

test('approve updateQuality', () => {
    function doUpdateQuality(name, sellIn, quality) {
        const items = [new Item(name, sellIn, quality)];
        const gildedRose = new GildedRose(items);
        gildedRose.updateQuality();
        return gildedRose.items;    
    }

    const names = ['', 'Aged Brie', 'Backstage passes to a TAFKAL80ETC concert', 'Sulfuras, Hand of Ragnaros'];
    const sellIns = [-1, 0, 1, 5, 6, 10, 11];
    const qualities = [0, 1, 49, 50];
    const params = [names, sellIns, qualities];

    const result = runCombinationSnapshot(params, doUpdateQuality);

    expect(result).toMatchSnapshot();
});

function runCombinationSnapshot(params, fn) {
    return product(params).map(combo => [combo, fn(...combo)]);
}

