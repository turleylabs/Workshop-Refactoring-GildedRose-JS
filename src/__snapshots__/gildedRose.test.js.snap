// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`approve updateQuality 1`] = `
Array [
  Array [
    Array [
      "",
      -1,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "",
      -1,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "",
      -1,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 47,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "",
      -1,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "",
      0,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "",
      0,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "",
      0,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 47,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "",
      0,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "",
      1,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "",
      1,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "",
      1,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "",
      1,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 49,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "",
      5,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "",
      5,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "",
      5,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "",
      5,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 49,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "",
      6,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "",
      6,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "",
      6,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "",
      6,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 49,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "",
      10,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "",
      10,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "",
      10,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "",
      10,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 49,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "",
      11,
      0,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "",
      11,
      1,
    ],
    Array [
      Item {
        "name": "",
        "quality": 0,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "",
      11,
      49,
    ],
    Array [
      Item {
        "name": "",
        "quality": 48,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "",
      11,
      50,
    ],
    Array [
      Item {
        "name": "",
        "quality": 49,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      -1,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      -1,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 3,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      -1,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      -1,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      0,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      0,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 3,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      0,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      0,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      1,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 1,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      1,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      1,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      1,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      5,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 1,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      5,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      5,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      5,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      6,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 1,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      6,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      6,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      6,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      10,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 1,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      10,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      10,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      10,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      11,
      0,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 1,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      11,
      1,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 2,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      11,
      49,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Aged Brie",
      11,
      50,
    ],
    Array [
      Item {
        "name": "Aged Brie",
        "quality": 50,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      -1,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      -1,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      -1,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      -1,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -2,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      0,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      0,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      0,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      0,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      1,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 3,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      1,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 4,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      1,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      1,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      5,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 3,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      5,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 4,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      5,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      5,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 4,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      6,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 2,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      6,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 3,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      6,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      6,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      10,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 2,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      10,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 3,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      10,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      10,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 9,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      11,
      0,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 1,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      11,
      1,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 2,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      11,
      49,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Backstage passes to a TAFKAL80ETC concert",
      11,
      50,
    ],
    Array [
      Item {
        "name": "Backstage passes to a TAFKAL80ETC concert",
        "quality": 50,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      -1,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      -1,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      -1,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      -1,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": -1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      0,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      0,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      0,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      0,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 0,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      1,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      1,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      1,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      1,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 1,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      5,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      5,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      5,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      5,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 5,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      6,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 6,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      6,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 6,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      6,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 6,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      6,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 6,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      10,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      10,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      10,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      10,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 10,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      11,
      0,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 0,
        "sellIn": 11,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      11,
      1,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 1,
        "sellIn": 11,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      11,
      49,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 49,
        "sellIn": 11,
      },
    ],
  ],
  Array [
    Array [
      "Sulfuras, Hand of Ragnaros",
      11,
      50,
    ],
    Array [
      Item {
        "name": "Sulfuras, Hand of Ragnaros",
        "quality": 50,
        "sellIn": 11,
      },
    ],
  ],
]
`;
